### ISSUE TITLE
Add string matching Algorithms for multiple programming languages. 

### ISSUE THREAD

**Issue by thecodershub** [thecodershub/algorithms#98]
- Simple substring check
- KMP algorithm
- Z algorithm

Please mention programming language and algorithm you're going to work in your claim here. Mention issue in pull request.

**Comment by DheerendraRathor** [comment]
I'm going to work on Z-algo in python

**Comment by NeedEspresso** [comment]
I'll work on simple substring check in python

**Comment by shrawnz** [comment]
I'll work on simple substring check in c++

**Comment by vaarsha** [comment]
I will work on simple substring check in c

**Comment by thecodershub** [comment]
I'll also work on KMP in Python

**Comment by thecodershub** [comment]
To all the contributors, please add tests and read the contribution guide before sending PRs.

**Comment by ManrajGrover** [comment]
Working on Simple Substring Check for Php, JavaScript and Java.

**Comment by ManrajGrover** [comment]
@thecodershub Are we allowed to use in built functions?

**Comment by thecodershub** [comment]
@ManrajGrover Yes. In that case add multiple methods - one with built in function and one without them.

**Comment by dnsaoki2** [comment]
I'll work on simple substring check in go

**Comment by claspina** [comment]
I'll work on simple substring check in kotlin.

**Comment by mkfeuhrer** [comment]
I'll work on Z algorithm in cpp

**Comment by Epsiloni** [comment]
Can I do string compare in Swift?
